lnclient/lnd/lnd.go

```diff
@@ -448,14 +415,104 @@ func NewLNDService(ctx context.Context, lndAddress, lndCertHex, lndMacaroonHex s
 		return nil, err
 	}
 
-	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey}
+	lndCtx, cancel := context.WithCancel(ctx)
+
+	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey, cancel: cancel}
+
+	// Subscribe to payments
+	go func() {
+		for {
+			paymentStream, err := lndClient.SubscribePayments(lndCtx, &routerrpc.TrackPaymentsRequest{
+				NoInflightUpdates: true,
+			})
+			if err != nil {
+				logger.Logger.WithError(err).Error("Error subscribing to payments")
+				continue
+			}
+			for {
+				select {
+				case <-lndCtx.Done():
+					return
+				default:
+					payment, err := paymentStream.Recv()
+					if err != nil {
+						logger.Logger.WithError(err).Error("Failed to receive payment")
+						time.Sleep(2 * time.Second)
```

Comment:
Why do we sleep here?

Reply:
Just to wait if that changes anything, otherwise we might keep looping into errors for some time